subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -0,0 +1,1212 @@
+            }
+        }
+
+        Ok(())
+    }
+
+    fn sync_txs(&mut self, history_txs_id: &HashSet<Txid>) -> Result<bool, Error> {
+        let mut txs_in_db = self.db.get_all_txid()?;
+        let txs_to_download: Vec<&Txid> = history_txs_id.difference(&txs_in_db).collect();
+        if !txs_to_download.is_empty() {
+            let mut batch = Batch::default();
+
+            let txs_bytes_downloaded = self.client.batch_transaction_get_raw(txs_to_download)?;
+            let mut txs_downloaded: Vec<BETransaction> = vec![];
+            for vec in txs_bytes_downloaded {
+                txs_downloaded.push(BETransaction::deserialize(&vec, self.network.id())?);
+            }
+            info!("txs_downloaded {:?}", txs_downloaded.len());
+            let mut previous_txs_to_download = HashSet::new();
+            for tx in txs_downloaded.iter() {
+                //self.db.insert_tx(&tx.txid(), &tx)?;
+                batch.insert(self.db.encrypt(tx.txid()), self.db.encrypt(tx.serialize()));
+                txs_in_db.insert(tx.txid());
+                for txid in tx.previous_output_txids() {
+                    previous_txs_to_download.insert(txid);
+                }
+
+                //TODO compute OutPoint Unblinded if tx is mine and it is liquid
+                if let BETransaction::Elements(tx) = tx {
+                    info!("compute OutPoint Unblinded");
+                    for (i, output) in tx.output.iter().enumerate() {
+                        if self.db.is_mine(&output.script_pubkey) {
+                            let txid = tx.txid();
+                            let vout = i as u32;
+                            let outpoint = elements::OutPoint {
+                                txid,
+                                vout,
+                            };
+                            if self.try_unblind(outpoint, output.clone()).is_err() {
+                                info!("{} cannot unblind, ignoring (could be sender messed up with the blinding process)", outpoint);
+                            }
+                        }
+                    }
+                }
+            }
+
+            let txs_to_download: Vec<&Txid> =
+                previous_txs_to_download.difference(&txs_in_db).collect();
+            if !txs_to_download.is_empty() {
+                let txs_bytes_downloaded =
+                    self.client.batch_transaction_get_raw(txs_to_download)?;
+                let mut txs_downloaded: Vec<BETransaction> = vec![];
+                for vec in txs_bytes_downloaded {
+                    txs_downloaded.push(BETransaction::deserialize(&vec, self.network.id())?);
+                }
+                info!("previous txs_downloaded {:?}", txs_downloaded.len());
+                for tx in txs_downloaded.iter() {
+                    //self.db.insert_tx(&tx.txid(), tx)?;
+                    batch.insert(self.db.encrypt(tx.txid()), self.db.encrypt(tx.serialize()));
```

Comment:
It would be more efficient if this kept just an `(spk,value,asset)` tuple of spent prev outpoints (plus the stuff needed for liquid), rather than the whole transaction. But I probably wouldn't consider this high priority.

Reply:
I think there are other things needed around like scripts, in the end a tx is pretty compact, especially if stripped from segwit like is done in the following MR